designs/2019-additional-lint-targets/README.md

```diff
@@ -0,0 +1,133 @@
+- Start Date: 2019-05-12
+- RFC PR:
+- Authors: Toru Nagashima ([@mysticatea](https://github.com/mysticatea))
+
+# Configuring Additional Lint Targets with `.eslintrc`
+
+## Summary
+
+This proposal adds the ability to specify additional target files into configuration files. This enhancement will solve the pain that people have to use the `--ext` option with wanted file extensions even if they use plugins which support additional file types.
+
+## Motivation
+
+People have to use the `--ext` option or glob patterns to check wanted files even if they use plugins which support additional file types.
+
+```yml
+plugins:
+  - markdown
+  - html
+  - "@typescript-eslint"
+  - react
+  - vue
+```
+
+```bash
+# ESLint checks only `*.js`.
+eslint src docs
+
+# Needs `--ext` option
+eslint src docs --ext .js,.md,.html,.ts,.jsx,.tsx,.vue
+```
+
+However, using plugins which support additional file types is the intention that wants to check those files. The requirement of redundant CLI options is not reasonable.
+
+Per "[Related Discussions](#related-discussions)" section, this is very regularly requested; I want to configure additional target files with `.eslintrc`.
+
+## Detailed Design
+
+This proposal enhances `overrides` property of the config file.
+
+- If a config file in a directory has `overrides` property, ESLint checks the files which are matched by any of override entries (i.e., `files`/`excludedFiles` criteria) additionally in the directory.
+    - If any of `files` value of a override entry ended with `*`, ESLint ignores the override entry to avoid checking all files.
+    - This enhancement affects only the directory path. If people gave glob patterns, ESLint behaves the same as currently.
```

Comment:
```suggestion
    - This enhancement affects only the case where a directory path is provided on the CLI. If people provide glob patterns on the CLI, ESLint behaves the same as currently.
```

I was a bit confused when reading this because I thought it meant glob patterns in `overrides`, not glob patterns on the CLI.